electrum/wallet.py

```diff
@@ -2033,13 +2034,20 @@ def is_address_reserved(self, addr: str) -> bool:
 
     def set_reserved_state_of_address(self, addr: str, *, reserved: bool) -> None:
         if not self.is_mine(addr):
+            # silently ignore non-ismine addresses
             return
         with self.lock:
+            has_changed = (addr in self._reserved_addresses) != reserved
             if reserved:
                 self._reserved_addresses.add(addr)
             else:
                 self._reserved_addresses.discard(addr)
-            self.db.put('reserved_addresses', list(self._reserved_addresses))
+            if has_changed:
+                self.db.put('reserved_addresses', list(self._reserved_addresses))
```

Comment:
We could remove this whole data structure from the wallet DB. With this change I am effectively generating it on startup anyway, so we don't need the persistence.

Reply:
I' always for removing things :-)